uaclient/contract_data_types.py

```diff
@@ -0,0 +1,572 @@
+from typing import List, Optional
```

Comment:
@lucasmoura @renanrodrigo please double check my opinions commented in this file